src/platform_impl/windows/monitor.rs

```diff
@@ -170,4 +175,44 @@ impl MonitorHandle {
     pub fn hidpi_factor(&self) -> f64 {
         self.hidpi_factor
     }
+
+    #[inline]
+    pub fn video_modes(&self) -> impl Iterator<Item = VideoMode> {
+        // EnumDisplaySettingsExW can return duplicate values (or some of the
+        // fields are probably changing, but we aren't looking at those fields
+        // anyway), so we're using a HashSet deduplicate
+        let mut modes = HashSet::new();
+        let mut i = 0;
+
+        loop {
+            unsafe {
+                let mut mode: wingdi::DEVMODEW = mem::zeroed();
+                mode.dmSize = mem::size_of_val(&mode) as WORD;
+                if winuser::EnumDisplaySettingsExW(
+                    self.monitor_info.szDevice.as_ptr(),
+                    i,
+                    &mut mode,
+                    0,
+                ) == 0
```

Comment:
Multi-line `if` clauses harm code readability. Could you pull this into a separate variable?

Reply:
I've moved `device_name` into a separate variable so this fits into a single line.